File: app/src/main/java/org/mifos/mobilebanking/api/services/UserDetailsService.java
package org.mifos.mobilebanking.api.services;

/*
 * Created by saksham on 13/July/2018
 */

import org.mifos.mobilebanking.api.ApiEndPoints;
import org.mifos.mobilebanking.models.UpdatePasswordPayload;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.PUT;

public interface UserDetailsService {

    @PUT(ApiEndPoints.USER)
    Observable<ResponseBody> updateAccountPassword(@Body UpdatePasswordPayload payload);
}

File: core/network/src/main/java/org/mifos/mobile/core/network/services/UserDetailsService.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-mobile/blob/master/LICENSE.md
 */
package org.mifos.mobile.core.network.services

import okhttp3.ResponseBody
import org.mifos.mobile.core.common.ApiEndPoints
import org.mifos.mobile.core.model.entity.UpdatePasswordPayload
import retrofit2.http.Body
import retrofit2.http.PUT

interface UserDetailsService {
    @PUT(ApiEndPoints.USER)
    suspend fun updateAccountPassword(@Body payload: UpdatePasswordPayload?): ResponseBody
}
